### PR TITLE
Cloning the `LoginViewModel` domain to `Onboarding` 

### PR DESCRIPTION
**Pull request by ouchadam**
Depends on #4656 Part of #4584

Nothing too exciting here, clones the `LoginViewModel`, state, actions and view events so that the Onboarding domain can override the behaviour and start to merge with `Login2`, will help reduce for the diff for the https://github.com/vector-im/element-android/pull/4668

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/ftue/FTUEAction.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/ftue/FTUEAction.kt:28]: I think there is a small misunderstanding here, FTUE stands for First Time User Experience, and it does not only cover the login flow, but also the first room creation, the first message sent, etc. So to me the FTUE is the user experience during the (2 or 3) first days using the app.

Also user can create an account or login even if they already know the app (so it will not be the first time)

So using FTUE to name classes related to the signin/signup flow is a bit strange to me.

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/ftue/FTUEAction.kt:28]: good point, I was considering the FTUE to be the new variant of the onboarding but product/design is calling this the `FTUE authentication flow`

what do you think of using `onboarding` as the package/activity/domain which would then contain `Login, Login2, FTUE Auth`?

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/ftue/FTUEAction.kt]: renamed ftue to onboarding https://github.com/vector-im/element-android/pull/4657/commits/119140b80e9ae61f80fc7eb7242cf8d246d06ce0

we have a onboarding domain with variants `legacy`, `login 2` and `ftue auth`

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/ftue/FTUEAction.kt]: Yes, Onboarding is better, also good for de-jargoning.
By the way, the Java recommendation for acronyms in naming is to capitalize only the first letter of them, so for instance we would have `FtueAction` here, instead of `FTUEAction`.

File: vector/src/main/java/im/vector/app/core/di/MavericksViewModelModule.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/core/di/MavericksViewModelModule.kt:454]: Maybe rename the fun to `onBoardingViewModelFactory`

**Review by bmarty — APPROVED** [review]
LGTM, just a small remark, but feel free to fix in another PR.
You can merge this one if you want to.

File: vector/src/main/java/im/vector/app/core/di/MavericksViewModelModule.kt
Comment (ouchadam) [reply vector/src/main/java/im/vector/app/core/di/MavericksViewModelModule.kt:454]: will update in the next PR to avoid having to wait for the CI to finish twice :smile:

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/core/di/MavericksViewModelModule.kt:454]: 9ccad11642ec00b67fcf47ac383bc4f9b5705ff1

**Pull request merged**
